src/lib/installationVerification.ts

```diff
@@ -358,117 +362,87 @@ export class InstallationVerification implements Verifier {
    */
   private async retrieveNpmMeta(): Promise<NpmMeta> {
     const logger = await this.getLogger();
-    return new Promise<NpmMeta>((resolve, reject) => {
-      const npmRegistry = getNpmRegistry();
-
-      logger.debug(`retrieveNpmMeta | npmRegistry: ${npmRegistry.href}`);
-      logger.debug(`retrieveNpmMeta | this.pluginNpmName.name: ${this.pluginNpmName.name}`);
-      logger.debug(`retrieveNpmMeta | this.pluginNpmName.scope: ${this.pluginNpmName.scope}`);
-      logger.debug(`retrieveNpmMeta | this.pluginNpmName.tag: ${this.pluginNpmName.tag}`);
-
-      if (this.pluginNpmName.scope) {
-        npmRegistry.pathname = path.join(
-          npmRegistry.pathname,
-          `@${this.pluginNpmName.scope}%2f${this.pluginNpmName.name}`
-        );
-      } else {
-        npmRegistry.pathname = path.join(npmRegistry.pathname, this.pluginNpmName.name);
-      }
-      logger.debug(`retrieveNpmMeta | npmRegistry.pathname: ${npmRegistry.pathname}`);
+    const npmRegistry = getNpmRegistry();
+
+    logger.debug(`retrieveNpmMeta | npmRegistry: ${npmRegistry.href}`);
+    logger.debug(`retrieveNpmMeta | this.pluginNpmName.name: ${this.pluginNpmName.name}`);
+    logger.debug(`retrieveNpmMeta | this.pluginNpmName.scope: ${this.pluginNpmName.scope}`);
+    logger.debug(`retrieveNpmMeta | this.pluginNpmName.tag: ${this.pluginNpmName.tag}`);
+
+    const npmShowModule = this.pluginNpmName.scope
+      ? `@${this.pluginNpmName.scope}/${this.pluginNpmName.name}`
+      : this.pluginNpmName.name;
+
+    const npmMetadata = new NpmModule(npmShowModule).show(npmRegistry.href);
+    const meta: NpmMeta = new NpmMeta();
```

Comment:
@maggiben rather than transferring data from NpmModule instance (or other locations) into NpmMeta, it seems that the NpmModule instance can be used as the source for data about the module. This would allow NpmMeta to be removed or consolidated into NpmModule.

Reply:
I took the class away and merged `NpmMeta` as a public prop in `NpmModule` class